### ISSUE TITLE
[Bug]: `rspress dev` does not update navbar item title when changing H1 inside doc

### ISSUE THREAD

**Issue by o-l-a-v** [web-infra-dev/rspress#1439]
### Version

* System: Windows 11 23H2
* Browser: Google Chrome
* RSPress v1.31.1

### Details

* Run rspress live with `pnpm rspress dev`
* Change `# Title` of a doc, say from "Title" to "Title 2". Save.
* New title is not reflected to navbar.

![Image](https://github.com/user-attachments/assets/a90bdf0d-98e2-4ec9-987d-ac93ca40956f)

Fix:

1 CTRL + C to stop rspress live view.
1. `pnpm rspress dev` to start live view again.
1. New title is reflected in navbar.

### Reproduce link

See details

### Reproduce Steps

See details

**Comment by Timeless0911** [comment]
Yeah, we currently only support sidebar HMR when we change `_meta.json`, so you can config the overview page title in `label` option at `_meta.json`.

```json
{
  "type": "file",
  "name": "index",
  "label": "API Overview"
}
```

**Comment by SoonIter** [comment]
fixed in https://github.com/web-infra-dev/rspress/pull/2549
